### PR TITLE
humble beginnings of a proper keptAt field for keeps

### PR DESCRIPTION
**Pull request by stkem**
@andrewconner @leogrim 
the do not merge is mostly because it's Friday afternoon and this touches the keep object and table

### CODE REVIEW

**Comment by stkem** [comment]
The keep object as well as the raw keep object gained a separate optional timestamp and the bookmark imports sets it when available. Next steps:
1) Backfill DB and make it not optional for keeps (it will remain optional for raw_keeps) 
2) Start using it everywhere appropriate instead of created at (that _might_ need extra indices on the DB)

**Comment by leogrim** [comment]
:+1:

**Comment by ymatsuda** [comment]
Keep me in the loop. This affects the search backend, right?

**Comment by leogrim** [comment]
We need to switch the field we use in the `KeepIndexer` from `createdAt` to `keptAt`, I can't think of anything else?

**Comment by andrewconner** [comment]
Awesome, thanks for this.

File: kifi-backend/modules/common/app/com/keepit/model/Keep.scala [hunk 1]
```diff
@@ -84,11 +85,11 @@ object Keep {
...
-    Keep(id, createdAt, updatedAt, externalId, title, uriId, isPrimary.exists(b => b), inDisjointLib.exists(b => b), urlId, url, bookmarkPath, visibility.getOrElse(isPrivateToVisibility(isPrivate)), userId, state, source, kifiInstallation, seq, libraryId)
+  def applyFromDbRow(id: Option[Id[Keep]], createdAt: DateTime, updatedAt: DateTime, externalId: ExternalId[Keep], title: Option[String], uriId: Id[NormalizedURI], isPrimary: Option[Boolean], inDisjointLib: Option[Boolean], urlId: Id[URL], url: String, bookmarkPath: Option[String], isPrivate: Boolean, userId: Id[User], state: State[Keep], source: KeepSource, kifiInstallation: Option[ExternalId[KifiInstallation]], seq: SequenceNumber[Keep], libraryId: Option[Id[Library]], visibility: Option[LibraryVisibility], keptAt: Option[DateTime]) = {
+    Keep(id, createdAt, updatedAt, externalId, title, uriId, isPrimary.exists(b => b), inDisjointLib.exists(b => b), urlId, url, bookmarkPath, visibility.getOrElse(isPrivateToVisibility(isPrivate)), userId, state, source, kifiInstallation, seq, libraryId, keptAt)
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/model/Keep.scala:89]: how about `Some(keptAt.getOrElse(createdAt))`?

**Pull request merged**